.github/workflows/ci.yml

```diff
@@ -147,10 +146,17 @@ jobs:
             -n 3 --maxfail 3 --durations 10 tests/unit tests/functional
             --force-flaky --no-flaky-report
 
-      # hooksample is installed as part of tests/requirements-libraries.txt, so run this step only if we
-      # installed all libraries.
+      # Install and test PyInstaller Hook Sample, to ensure that tests declared in
+      # entry-points are discovered.
+      - name: Install hooksample
+        run: pip install "https://github.com/pyinstaller/hooksample/archive/v4.0rc1.zip"
+
+      - name: Inject bogus hook
+        run: |
+          hooks_repo_dir="$(python -c 'from _pyinstaller_hooks_contrib.hooks.stdhooks import __path__; print(__path__[0])')"
+          echo 'assert 0, "Wrong hook! Use the pyi_hooksample copy instead!"' > "$hooks_repo_dir/hook-pyi_hooksample.py"
```

Comment:
I suspect it will be less painful to write the hook file in that python command as well. 

Otherwise, we'll need `shell: bash` for Windows, and that raises the question of which python will be in `PATH` and whether the path will be in "correct" format (Windows one or the mingw/msys2 one)...

Reply:
`shell: bash` works mostly fine on Windows. `PATH` is set up properly. Commands such as `which` return `/c/msys2/style/paths` and you have to avoid using backslashes in paths for fear of them being them misinterpreted as escapes.